### ISSUE TITLE
bug 🐛: every sl command crashes with "unknown python exception"

### ISSUE THREAD

**Issue by vegerot** [facebook/sapling#960]
(almost) Every sl command I run gives the same error message:

```sh
$ sl
unknown python exception

$ sl show
unknown python exception

$ sl --version
Sapling 4.4.2_20240930_132004_4a1d4f6f8e0d
(see https://sapling-scm.com/ for more information)

$ CHGDISABLE=1 sl
unknown python exception
```

I bisected Sapling and found the culprit: 4a1d4f6f8e0de7b33ea060161e0c4d19e6178112
I can confirm that Python 3.11 works fine but 3.12 💥

This is on macOS 14.7 and Python 3.12.6 from homebrew

**Comment by vegerot** [comment]
(tagging #957)

**Comment by ahornby** [comment]
https://github.com/facebook/sapling/pull/964 might help.  I got the tests up and running and then used them to validate 3.12 on Ubuntu 24.04

**Comment by markbhasawut** [comment]
This issue should be fixed by commit 65a7e9097fb9280aef7c50ecdf08b5755288490a. I was able to run sl commands on macOS with Python 3.12.7 and successfully validated the tests without any exceptions.

**Comment by vegerot** [comment]
@markbhasawut Confirmed!  Thanks
